cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -149,6 +151,9 @@ public FieldLineageInfo(Collection<? extends Operation> operations) {
 
     this.operations = new HashSet<>(operations);
     this.checksum = computeChecksum();
+    this.destinationFields = computeDestinationFields();
```

Comment:
computing these can be expensive. Perhaps they should be computed on demand only if required? That means, however, need to make sure they are computed before serializing for storing it.